### DISCUSSION TITLE
Unrecognized pair style 'deepmd' (src/force.cpp:279)

### DISCUSSION THREAD

**Discussion by chenfan22** [deepmodeling/dpgen#1151] [Q&A]
Dear All,

I got the following error message from the log file when I tried to run the step2 example code for CH4 ( dpgen run param.json machine.json > log.out 2>&1 ):
![image](https://user-images.githubusercontent.com/100225001/224384043-021d87fd-bb8c-4798-beae-b7bfff351620.png)

The log.lammp file from that folder shows an error message:
ERROR: Unrecognized pair style 'deepmd' (src/force.cpp:279)
Last command: pair_style      deepmd ../graph.000.pb ../graph.003.pb ../graph.002.pb ../graph.001.pb  out_freq ${THERMO_FREQ} out_file model_devi.out

dpgen.log file is shown belowed:
![image](https://user-images.githubusercontent.com/100225001/224384625-d158ea41-bea2-4373-8c4d-f1f3d6d9680c.png)

The version of deepmd-kit is 2.1.5 which was installed using "conda create -n deepmd deepmd-kit=*=*cpu libdeepmd=*=*cpu lammps -c https://conda.deepmodeling.com -c defaults". 

I also check lmp -h (lmp is under ~/miniconda3/envs/deepmd/bin/lmp):
* Pair styles:

adp             agni            airebo          airebo/morse    atm
beck            body/nparticle  body/rounded/polygon
body/rounded/polyhedron         bop             born            born/coul/dsf
born/coul/dsf/cs                born/coul/long  born/coul/long/cs
born/coul/msm   born/coul/wolf  born/coul/wolf/cs               brownian
brownian/poly   buck            buck/coul/cut   buck/coul/long
buck/coul/long/cs               buck/coul/msm   buck/long/coul/long
buck/mdf        colloid         comb            comb3           cosine/squared
coul/cut        coul/cut/global coul/cut/soft   coul/debye      coul/diel
coul/dsf        coul/exclude    coul/long       coul/long/cs    coul/long/soft
coul/msm        coul/slater/cut coul/slater/long                coul/streitz
coul/wolf       coul/wolf/cs    reax            dsmc            e3b
eam             eam/alloy       eam/alloy/opt   eam/cd          eam/cd/old
eam/fs          eam/fs/opt      eam/he          eam/opt         edip
edip/multi      eim             extep           gauss           gauss/cut
gayberne        gran/hertz/history              gran/hooke
gran/hooke/history              granular        gw              gw/zbl
harmonic/cut    hbond/dreiding/lj               hbond/dreiding/morse
hybrid          hybrid/overlay  hybrid/scaled   lcbop           lennard/mdf
line/lj         list            lj96/cut        lj/charmm/coul/charmm
lj/charmm/coul/charmm/implicit  lj/charmm/coul/long
lj/charmm/coul/long/opt         lj/charmm/coul/long/soft
lj/charmm/coul/msm              lj/charmmfsw/coul/charmmfsh
lj/charmmfsw/coul/long          lj/class2       lj/class2/coul/cut
lj/class2/coul/cut/soft         lj/class2/coul/long
lj/class2/coul/long/cs          lj/class2/coul/long/soft        lj/class2/soft
lj/cubic        lj/cut          lj/cut/coul/cut lj/cut/coul/cut/soft
lj/cut/coul/debye               lj/cut/coul/dsf lj/cut/coul/long
lj/cut/coul/long/cs             lj/cut/coul/long/opt
lj/cut/coul/long/soft           lj/cut/coul/msm lj/cut/coul/wolf
lj/cut/dipole/cut               lj/cut/dipole/long              lj/cut/opt
lj/cut/soft     lj/cut/tip4p/cut                lj/cut/tip4p/long
lj/cut/tip4p/long/opt           lj/cut/tip4p/long/soft          lj/expand
lj/expand/coul/long             lj/gromacs      lj/gromacs/coul/gromacs
lj/long/coul/long               lj/long/coul/long/opt
lj/long/dipole/long             lj/long/tip4p/long              lj/mdf
lj/relres       lj/sf/dipole/sf lj/smooth       lj/smooth/linear
lj/sf           local/density   lubricate       lubricateU      lubricateU/poly
lubricate/poly  meam            meam/c          meam/spline     meam/sw/spline
mie/cut         momb            morse           morse/opt
morse/smooth/linear             morse/soft      nb3b/harmonic   nm/cut
nm/cut/coul/cut nm/cut/coul/long                nm/cut/split    peri/eps
peri/lps        peri/pmb        peri/ves        polymorphic     reaxff
reax/c          rebo            resquared       soft            srp
sw              sw/angle/table  sw/mod          table           tersoff
tersoff/mod     tersoff/mod/c   tersoff/table   tersoff/zbl     threebody/table
tip4p/cut       tip4p/long      tip4p/long/soft tracker         tri/lj
ufm             ufm/opt         vashishta       vashishta/table wf/cut
yukawa          yukawa/colloid  zbl             zero

I didn't find deepmd pair styles from lmp -h. I reinstalled deepmd-kit with version 2.1.1 and 2.1.4. They still have the same problem.
Could you please give me some suggestion ? Thank you in advance for your help.

**Comment by njzjz** [comment]
See https://github.com/deepmodeling/deepmd-kit/issues/1809.

**Reply by chenfan22** [reply]
Thanks for the reply.  I actually ran it in an activated deepmd environment. An extra env file provided for the source_list command from machine file solves this problem.

**Reply by chb-deep** [reply]
I also had the same err, how did you do it specifically

**Reply by niuniu800** [reply]
> Thanks for the reply. I actually ran it in an activated deepmd environment. An extra env file provided for the source_list command from machine file solves this problem.

Could you please let me know how you set up the activation environment in the source_list file within the machine directory? I've encountered the same issue as you.
